adoc/extensions/sycl_khr_group_interface.adoc

```diff
@@ -0,0 +1,849 @@
+[[sec:khr-group-interface]]
+= sycl_khr_group_interface
+
+This extension provides an alternative interface for groups of work-items
+(including work-groups, sub-groups, and individual work-items) that is simpler
+and less verbose than the interface provided by `sycl::group` and
+`sycl::sub_group` in SYCL 2020.
+
+[[sec:khr-group-interface-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+Some features of this extension are only available when a SYCL implementation
+conforms to {cpp23} or later.
+
+[[sec:khr-group-interface-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_GROUP_INTERFACE# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+[[sec:khr-group-interface-common]]
+== Common group interface
+
+The `khr::work_group`, `khr::sub_group` and `khr::member_item` objects defined
+by this extension all implement a common set of operations, which are shown in
+the synopsis below.
+The name [code]#+__group__+# in that synopsis is a placeholder for each of these
+three types.
+When the synopsis shows an ellipsis ([code]#+/*...*/+#), the subsequent sections
+clarify the definition for each type.
+
+Whether it is possible to query certain properties of a group at compile-time
+depends on the group's type.
+A member function that is not declared [code]#constexpr# in the synopsis below
+may still be declared [code]#constexpr# for a specific group type defined in a
+subsequent section.
+
+[source,role=synopsis]
+----
+
+namespace sycl::khr {
+
+class __group__ {
+
+ public:
+  using id_type = /* ... */;
+  using linear_id_type = /* ... */;
+  using range_type = /* ... */;
+  using extents_type = /* ... */; // C++23
+  using size_type = /* ... */;
+  static constexpr int dimensions = /* ... */;
+  static constexpr memory_scope fence_scope = /* ... */;
+
+  id_type id() const noexcept;
+  linear_id_type linear_id() const noexcept;
+
+  range_type range() const noexcept;
```

Comment:
There's no `linear_range()` query, is this replaced with `size()`?

Reply:
No. `range()` returns the range that the `id` is drawn from (i.e., the number of groups), whereas `size()` returns the size of the group (i.e., the number of work-items).

We didn't add `linear_range()` because it's shorter to write `range.size()`.

Reply:
That makes sense thanks.